performant-software/Annotation-Studio
# Improve user experience for selecting which users annotation to see

## What does this PR do?
1. Clicking Me/Teams/Editions should set the User filter menu to "All Users", and should issue a query to the API with mode=group for Teams or mode=class for Editions (or mode=user, ONLY if "Me" is clicked).
2. Selecting a specific User from the User filter menu should issue a request to the API with mode=user and user=[email-of-user-selected], and unselect all Me/Teams/Editions filter buttons above.
3. Selecting "All Users" User from the User filter menu should issue a request to the API with mode=class, and set the "Editions" button to selected. This is identical to selecting "Editions" from filter button menu.

## How to test?
First Case:
1. Login and go to https://staging.covecollective.org/documents/lady-audleys-secret-installment-2
2. Click on the Me/Teams/Editions then that should set the User filter menu to "All Users".
3. Select any user from "By User" dropdown list.
4. Again click on the Me/Teams/Editions then that should set the User filter menu to "All Users".
5. Click on the Me/Teams/Editions then select any user from "By User" dropdown option list.
6. Now you shouldn't see any of the Me/Teams/Editions buttons selected
